ha/core/event_manager/event_manager.py

```diff
@@ -317,10 +317,18 @@ def subscribe(self, component: SUBSCRIPTION_LIST, events: List[SubscribeEvent])
             message_type = self._create_message_type(component)
             for event in events:
                 subscription_key = EVENT_MANAGER_KEYS.SUBSCRIPTION_KEY.value.replace("<component_id>", component)
-                self._store_component_key(subscription_key, event.resource_type, event.states)
-                self._store_event_key(event.resource_type, event.states, comp=component)
-                for state in event.states:
-                    self._monitor_rule.add_rule(event.resource_type, state, self._default_action)
+                if event.functional_types:
+                    for func_type in event.functional_types:
+                        resource_type = event.resource_type + HA_DELIM + func_type
```

Comment:
can you add a comment on how resource_type will look, for better understanding.

Reply:
Done.